src/pageObject.js

```diff
@@ -1,16 +1,44 @@
 'use strict';
 
 class Header {
+  constructor(logo = 'Conduit logo') {
+    this.logo = logo;
+  }
 
-};
+  getLogo() {
+    return this.logo;
+  }
+}
 
 class PageObject {
+  constructor(url, header = new Header()) {
+    this.url = url;
+    this.header = header;
+  }
 
-};
+  clickOnLogo() {
+    return `Click on the ${this.header.getLogo()}`;
+  }
+}
 
 class ArticlePage extends PageObject {
+  constructor(url, slug, header = new Header()) {
+    super(url, header);
+    this.fullUrl = `${url}${slug}`;
```

Comment:
The `fullUrl` property is a derived property, it's better to calculate it on the fly in a getter method instead of storing it. This way it will always be up-to-date even if `url` or `slug` change.